amethyst_derive/tests/test.rs

```diff
@@ -75,3 +75,15 @@ pub struct Outer {
     #[prefab(Component)]
     external: External,
 }
+
+#[derive(PrefabData, Clone)]
```

Comment:
Since the test prefab data was switched over to only struct variants, will enum tuples and regular enums be supported by this code? Can all the different variant types get added to the tests and prefab example?

Reply:
The existing code only supported named fields and I didn't change that so this only supports struct variants. My intention was to leave it at that but I think I see how you could support tuple variants (and tuple structs as well), it would add a bit of complexity but probably isn't too bad. I actually hadn't thought about regular enums at all. Let me check that...

Reply:
Yeah, regular enums work fine. I've added a test for that.

Reply:
Ok. There's support for tuple structs and tuple enum variants. Wasn't too bad. I think that left the code a bit cluttered so I may come back and tidy up some but it works and there are tests for it on both the struct and enum side.